src/test/ui/argument-suggestions/two-mismatch-notes.stderr

```diff
@@ -4,14 +4,14 @@ error[E0308]: arguments to this function are incorrect
 LL |     foo(f, w);
    |     ^^^
    |
-note: expected `i32`, found `u32`
+note: expected fn pointer, found fn item
```

Comment:
Hm, I think this note isn’t particularly helpful. Even harmful perhaps. “Compatible” fn items can always coerce to fn pointers, but the note may impress upon the reader that this isn’t always the case for some reason. So there is definitely something more the compiler could say. I think the 2nd note is pretty good and could potentially stay as the only note in this error.

Reply:
Yeah, this is supposed to be "expected `fn(i32)`, found `fn(u32) {f}`", *but* I didn't modify the labels for the special cases (like this one) to keep the diff much shorter. I can go all out and apply the changes in more places to fix this one case, but it will dramatically increase the size of the PR.

Reply:
I think it should be ok to land this here, provided we later also have a structured suggestion for functions. I'm pretty sure this is not the only place where we are already having this unfortunate output and where we should be explicit with a note (which I thought we had) about that behavior.